src/main/java/uk/gov/pay/ledger/queue/EventMessageHandler.java

```diff
@@ -61,6 +65,7 @@ private void processSingleMessage(EventMessage message) throws QueueException {
         if(response.isSuccessful()) {
             EventDigest eventDigest = eventService.getEventDigestForResource(event.getResourceExternalId());
             transactionService.upsertTransactionFor(eventDigest);
+            transactionMetadataService.upsertMetadataFor(eventDigest);
```

Comment:
Don't think we want to process on `eventDigest`. 

Event digest includes all events for a given transaction. So with this, we will be processing metadata every time we get event for a transaction and could impact performance. We could only process metadata if event payload has `external_metadata` object